catalog/core/catalog-core-validator/src/main/java/ddf/catalog/validation/impl/ValidationParser.java

```diff
@@ -0,0 +1,328 @@
+                .endsWith(".json");
+    }
+
+    public void setAttributeRegistry(AttributeRegistry attributeRegistry) {
+        this.attributeRegistry = attributeRegistry;
+    }
+
+    public void setAttributeValidatorRegistry(
+            AttributeValidatorRegistry attributeValidatorRegistry) {
+        this.attributeValidatorRegistry = attributeValidatorRegistry;
+    }
+
+    @SuppressWarnings("unchecked")
+    private void parseValidators(Map<String, Object> root, Outer outer) {
+        if (root == null || root.get("validators") == null) {
+            return;
+        }
+
+        Map<String, List<Outer.Validator>> validators = new HashMap<>();
+        for (Map.Entry<String, Object> entry : ((Map<String, Object>) root.get("validators")).entrySet()) {
+            String rejson = JsonFactory.create()
+                    .toJson(entry.getValue());
+            List<Outer.Validator> lv = JsonFactory.create()
+                    .readValue(rejson, List.class, Outer.Validator.class);
+            validators.put(entry.getKey(), lv);
+        }
+        outer.validators = validators;
+    }
+
+    private List<Callable<Boolean>> parseAttributeTypes(
+            Map<String, Outer.AttributeType> attributeTypes) {
+        List<Callable<Boolean>> staged = new ArrayList<>();
+        for (Map.Entry<String, Outer.AttributeType> entry : attributeTypes.entrySet()) {
+            final AttributeDescriptor descriptor = new AttributeDescriptorImpl(entry.getKey(),
+                    entry.getValue().indexed,
+                    entry.getValue().stored,
+                    entry.getValue().tokenized,
+                    entry.getValue().multivalued,
+                    BasicTypes.getAttributeType(entry.getValue().type));
+
+            staged.add(() -> attributeRegistry.registerAttribute(descriptor));
+        }
+        return staged;
+    }
+
+    private List<Callable<Boolean>> parseMetacardTypes(List<Outer.MetacardType> metacardTypes) {
+        List<Callable<Boolean>> staged = new ArrayList<>();
+        Bundle bundle = FrameworkUtil.getBundle(this.getClass());
+        if (bundle == null) {
+            LOGGER.warn(String.format(
+                    "Unable to get bundle for [%s], Metacard Types will not be registered",
+                    this.getClass()
+                            .getName()));
+            return new ArrayList<>();
+        }
+        BundleContext context = bundle.getBundleContext();
+        for (Outer.MetacardType metacardType : metacardTypes) {
+            Set<AttributeDescriptor> attributeDescriptors =
+                    new HashSet<>(BasicTypes.BASIC_METACARD.getAttributeDescriptors());
+            Set<String> requiredAttributes = new HashSet<>();
+
+            for (Map.Entry<String, Outer.MetacardAttribute> entry : metacardType.attributes.entrySet()) {
+                attributeDescriptors.add(attributeRegistry.getAttributeDescriptor(entry.getKey())
+                        .get());
```

Comment:
Right. Do we want to want to stop parsing and throw an error if they require an attribute that hasn't been registered?